### PR TITLE
fix: replace long shebangs with `/usr/bin/env`

### PR DESCRIPTION
**Pull request by wolfv**
Note: I should test the whole replace function as well

### CODE REVIEW

**Comment by wolfv** [comment]
reference:

https://github.com/mamba-org/mamba/blob/b5195d96b062c841fcf6ead36a611b1cf4707359/libmamba/src/core/link.cpp#L101-L123

https://github.com/conda/conda/blob/c57fbdebed072ced5e03fabac13edcf3b0f444c9/conda/core/portability.py#L239-L240

https://github.com/conda/conda/blob/c57fbdebed072ced5e03fabac13edcf3b0f444c9/conda/core/portability.py#L67-L68

File: crates/rattler/src/install/link.rs [hunk 1]
```diff
@@ -497,6 +498,45 @@ pub fn copy_and_replace_placeholders(
     }
     Ok(())
 }
+use once_cell::sync::Lazy;
+
+static SHEBANG_REGEX: Lazy<Regex> = Lazy::new(|| {
```
Comment (baszalmstra) [inline crates/rattler/src/install/link.rs:503]: We really dont need a regex for this I think..

Comment (wolfv) [reply crates/rattler/src/install/link.rs:503]: The main complication is escaped white spaces in the path ...

File: crates/rattler/src/install/link.rs [hunk 2]
```diff
@@ -364,18 +366,10 @@ fn reflink_to_destination(
             Err(e) if e.kind() == ErrorKind::Unsupported && !allow_hard_links => {
                 return copy_to_destination(source_path, destination_path)
             }
-            Err(e) => {
+            Err(_) => {
                 return if allow_hard_links {
-                    tracing::debug!(
```
Comment (wolfv) [inline crates/rattler/src/install/link.rs:369]: removed these as they are super noisy in CI.

**Review by baszalmstra — APPROVED** [review]

**Pull request merged**